FreeRTOS-Plus/Demo/FreeRTOS_Cellular_Interface_Windows_Simulator/Common/cellular_setup.c

```diff
@@ -131,27 +127,55 @@ bool setupCellular( void )
 
             vTaskDelay( pdMS_TO_TICKS( CELLULAR_SIM_CARD_WAIT_INTERVAL_MS ) );
         }
+
+        if( cellularStatus != CELLULAR_SUCCESS )
+        {
+            configPRINTF( ( ">>>  Cellular SIM failure  <<<\r\n" ) );
+        }
+    }
+
+    /* Turn off PSM because this is demo to showcase MQTT instead of PSM mode. */
+    if( cellularStatus != CELLULAR_SUCCESS )
+    {
+        psmSettings.mode = 0;
+        cellularStatus = cellularStatus = Cellular_SetPsmSettings( CellularHandle, &psmSettings );
+
+        if( cellularStatus != CELLULAR_SUCCESS )
+        {
+            configPRINTF( ( ">>>  Cellular_SetPsmSettings failure  <<<\r\n" ) );
+        }
     }
 
     /* Setup the PDN config. */
     if( cellularStatus == CELLULAR_SUCCESS )
     {
         cellularStatus = Cellular_SetPdnConfig( CellularHandle, CellularSocketPdnContextId, &pdnConfig );
-    }
-    else
-    {
-        configPRINTF( ( ">>>  Cellular SIM failure  <<<\r\n" ) );
+
+        if( cellularStatus != CELLULAR_SUCCESS )
+        {
+            configPRINTF( ( ">>>  Cellular_SetPdnConfig failure %d  <<<\r\n", cellularStatus ) );
+        }
     }
 
     /* Rescan network. */
     if( cellularStatus == CELLULAR_SUCCESS )
     {
         cellularStatus = Cellular_RfOff( CellularHandle );
```

Comment:
Cycling RF On and Off should only be required when modifying the default EPS bearer (usually profile #1).

Reply:
The demo code tries not to assume previous PDN setting in case cellular modem can not register to the network.
Adding Cellular_GetPdnConfig API to cellular library is another approach to decide if RF On and Off is required.
Currently, it is not included in cellular interface API. We may discuss to add this feature.

Reply:
Yes. That would be a welcome improvement. Some networks throttle registration attempts or will fail device certification if extra registration events are detected.